realm/realm-library/src/main/java/io/realm/internal/Context.java

```diff
@@ -16,61 +16,90 @@
 
 package io.realm.internal;
 
-import java.lang.ref.Reference;
 import java.lang.ref.ReferenceQueue;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
 
 public class Context {
 
+    // Pool to hold the phantom references.
+    // The size of array for storing phantom references will never decrease. Instead, we use another array to hold the
```

Comment:
Perhaps also add a comment about the pool holding old values that are not deleted for efficiency reasons: https://github.com/realm/realm-java/pull/2525/files#diff-a5b83c6e9041e4c99a95e860f0c4dd54R100